### ISSUE TITLE
Unable to directly queue operations multiple times for tape transform

### ISSUE THREAD

**Issue by glassnotes** [PennyLaneAI/pennylane#1218]
#### Issue description

* *Context:* Trying to implement a tape transform that performs CNOT pair insertion, i.e., replacing every CNOT on a tape with multiple copies of that CNOT.

* *Expected behavior:*  When performing a tape transform, running `op.queue()` on an operation `op` multiple times, I expected it would add that operation to the queue multiple times. 

* *Actual behavior:* Operations are only queued a single time.
 
* *Reproduces how often:* Always

* *System information:* (post the output of `import pennylane as qml; qml.about()`)
```
Name: PennyLane
Version: 0.15.0.dev0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: None
Author-email: None
License: Apache License 2.0
Location: /home/olivia/Software/anaconda3/envs/qite/lib/python3.7/site-packages
Requires: appdirs, scipy, autograd, semantic-version, networkx, toml, numpy
Required-by: PennyLane-qiskit
Platform info:           Linux-5.4.0-71-generic-x86_64-with-debian-bullseye-sid
Python version:          3.7.10
Numpy version:           1.20.1
Scipy version:           1.6.1
Installed devices:
- default.gaussian (PennyLane-0.15.0.dev0)
- default.mixed (PennyLane-0.15.0.dev0)
- default.qubit (PennyLane-0.15.0.dev0)
- default.qubit.autograd (PennyLane-0.15.0.dev0)
- default.qubit.jax (PennyLane-0.15.0.dev0)
- default.qubit.tf (PennyLane-0.15.0.dev0)
- default.tensor (PennyLane-0.15.0.dev0)
- default.tensor.tf (PennyLane-0.15.0.dev0)
- qiskit.aer (PennyLane-qiskit-0.15.0.dev0)
- qiskit.basicaer (PennyLane-qiskit-0.15.0.dev0)
- qiskit.ibmq (PennyLane-qiskit-0.15.0.dev0)

```

#### Source code and tracebacks
```python
import pennylane as qml

ops = [
    qml.PauliX(wires=0),
    qml.Hadamard(wires=1),
    qml.CNOT(wires=[0, 1])
]

with qml.tape.QuantumTape() as some_tape:
    for op in ops:
        op.queue()
        op.queue()

>>> print(some_tape.operations)
[PauliX(wires=[0]), Hadamard(wires=[1]), CNOT(wires=[0, 1])]
```

A work-around is to create copies of the operations, but it feels clunky.

```python
with qml.tape.QuantumTape() as some_tape:
    for op in ops:
        op.queue()
        copy = op.__copy__()
        copy.queue()

>>> print(some_tape.operations)
[PauliX(wires=[0]), PauliX(wires=[0]), Hadamard(wires=[1]), Hadamard(wires=[1]), CNOT(wires=[0, 1]), CNOT(wires=[0, 1])]
```

**Comment by josh146** [comment]
Thanks for catching this @glassnotes! I'll have to digest - I agree with you that the expected behaviour is that calling `op.queue()` multiple times will re-queue.

However, I fear this is expected behaviour, since an object can only exist in the queue once (and merely the annotations of that object in the queue are updated).

@co9olguy, any thoughts?

**Comment by glassnotes** [comment]
> However, I fear this is expected behaviour

Haha I had a feeling that this was the case :sweat_smile: 

Would it make sense to have an `op.queue_multiple()`, or pass an argument like the below?

```python
with qml.tape.QuantumTape() as some_tape:
    for op in ops:
        op.queue(copies=3)
```

That said, this is a pretty limited use case and probably something that can just be cleared up with documentation.

**Comment by co9olguy** [comment]
In general, I'd like us to get away from users or developers ever having to manage queueing. So my reaction here is less "it's a bug that we can't queue multiple times" but rather "why do we allow calling `op.queue`?"

**Comment by josh146** [comment]
> So my reaction here is less "it's a bug that we can't queue multiple times" but rather "why do we allow calling op.queue?"

The queuing ADR recommended added `qml.apply(op)` as a end-user replacement for `op.queue()`, we are definitely due for it.

**Comment by glassnotes** [comment]
This can be closed now that #1433 is merged :tada:  :tada:
